src/server/capio_server.cpp

```diff
@@ -207,16 +208,12 @@ int parseCLI(int argc, char **argv, int rank) {
     } else {
 #ifdef CAPIOLOG
         // log file not given. starting with default name
-        auto hostname = new char[HOST_NAME_MAX];
-        gethostname(hostname, HOST_NAME_MAX);
-
         const std::string logname =
-            CAPIO_LOG_SERVER_DEFAULT_FILE_NAME + std::string(hostname) + ".log";
+            CAPIO_LOG_SERVER_DEFAULT_FILE_NAME + std::to_string(capio_syscall(SYS_gettid)) + ".log";
```

Comment:
Here you are already passing the `tid` to the filename.
Please move this logic to #33 which is the correct place to discuss it